examples/common/utility/utility.hpp

```diff
@@ -356,6 +358,59 @@ class cli_argument_pack {
     }
 }; // class cli_argument_pack
 
+// utility class to aid relative error measurement of samples
+class measurements {
+public:
+    measurements() = default;
```

Comment:
I would like to keep this class default constructible. This allows flexibility in cases where the number of iterations is unknown in advance, or when the user chooses not to specify it. While this may lead to multiple vector allocations, which isn’t optimal, I believe it's an acceptable trade-off.